### DISCUSSION TITLE
[Feature Request] Allowing Edit in [Open Notebook] Dialog

### DISCUSSION THREAD

**Discussion by k4nzdroid** [zim-desktop-wiki/zim-desktop-wiki#2408] [Ideas]
There are three hidden options: "shared", ''end_of_line'', ''disable_trash''. They all have one thing in common: controlling how a notebook gets loaded. (''disable_trash' maybe not.)

Here is the ideal: add an [Edit] button in the [Open Notebook] dialog, allowing users to control how  a notebook gets loaded.

This ideal is inspired by the needing for persisting index.db and state.conf. Indexing 9456 files is quite a long while. :)

And maybe some related problems will get soloved, like bookmark persisting.

# Future more

Introduce the .zim folder anyway for the following reasons:
* #32 , If we want to support Rearrange Manually, there has to be some place to hold those configrations permanently.
* For myself, I do not like having any none-note files in the notebook folder. Yeah, the notebook.zim is what i am referring to. :)
* As the manual said: `Note that zim by design is a single-user program.`, I think it is fine to put the related data inside notebook.

**Comment by pfalcon** [comment]
> Here is the ideal: add an [Edit] button in the [Open Notebook] dialog, allowing users to control how a notebook gets loaded.

Excuse me, but how an **Edit** button may control how a something gets **loaded**? Edit button edits something.

**Reply by k4nzdroid** [reply]
The Edit button is used to open a config dialog for a specific notebook. In the config dialog, users can make choices to control how a notebook is loaded.

 Something like this
![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/40691275/3fa09098-b8ea-423c-85fa-76dd64cbaddb)

**Comment by pfalcon** [comment]
The likely the button should be named "Options..." or "Configure...".

**Reply by k4nzdroid** [reply]
Yep, you are right. "Edit" is a bad name.

**Comment by jaap-karssenberg** [comment]
My take on this:
- allow to set the options in the "Add" dialog as well
- put the options under an expander with "advanced" options, so regular
users feel safe to ignore these
- it is OK to put them in the regular notebook options dialog with a label
"requires re-start" -- see e.g. how it was done in the preferences dialog
for the window decoration option
- would need to do code check, but me guess is that only "shared" is really
fixes at start up, the other two options can be switched on run time
